doc/development_setup.md

```diff
@@ -0,0 +1,8 @@
+# Development Setup
+
+Operating Systems: 
+ - Windows
+ - macOS (Unix)
+
```

Comment:
Which browsers will you use?

Reply:
Also which add-ons etc. You can also extend the list later, you don't need to decide on everything right now.

We also should use rustfmt to have consistent formatting.

Btw. https://code.visualstudio.com/docs/languages/rust recommends rust-analyzer